### PR TITLE
Update dropall atcommand

### PR DESCRIPTION
**Pull request by Emistry**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- The correct amount of dropped items should be displayed.
- Included skipped amount.

**Issues addressed:** 
Merge from https://github.com/rathena/rathena/commit/beb6b579f21158411c885b19ff955ba4c49d5e71

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2545)

**Comment by Lemongrass3110** [comment]
Either I read this wrong or you might have merged it badly. You still access the count inside the inventory after it was deleted...

**Comment by dastgirp** [comment]
@Lemongrass3110 is correct

File: src/map/atcommand.c [hunk 1]
```diff
@@ -5380,25 +5379,31 @@ ACMD(dropall)
...
 				continue;
+			
 			if (type == -1 || type == item_data->type) {
 				if (sd->status.inventory[i].equip != 0)
 					pc->unequipitem(sd, i, PCUNEQUIPITEM_RECALC | PCUNEQUIPITEM_FORCE);
-				count += sd->status.inventory[i].amount;
-				pc->dropitem(sd, i, sd->status.inventory[i].amount);
+				
+				if (pc->dropitem(sd, i, sd->status.inventory[i].amount) != 0)
+					count += sd->status.inventory[i].amount;
```
Comment (dastgirp) [inline src/map/atcommand.c]: sd->status.inventory.amount should be saved in variable before and then added here.
Using it after dropping item can result in crash/garbage value.
See @Lemongrass3110 comment for more details

Comment (4144) [reply src/map/atcommand.c]: no crash because here no pointer, but look like it will be 0

Comment (Emistry) [reply src/map/atcommand.c]: fixed.

**Pull request merged**